File: two_sum/two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  const numMap = new Map();
  
  for (i in nums) {
      if (numMap.has(target - nums[i])) {
          return [i, numMap.get(target - nums[i])];
      };
      
      numMap.set(nums[i], i);
  }
};

File: two_sum/two_sum.go
func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for i, num := range nums {
		complement := target - num
		index, found := m[complement]
		if found {
			return []int{i, index}
		}

		m[num] = i
	}

	return make([]int, 0)
}